theme/lumo/vaadin-notification-styles.html

```diff
@@ -131,6 +131,66 @@
       :host([slot^="bottom"][closing]) {
         animation: lumo-notification-exit-slide-down 300ms;
       }
+
+      :host([theme~="primary"]) [part="overlay"] {
+        background: var(--lumo-primary-color);
+        color: var(--lumo-primary-contrast-color);
+      }
+
+      :host([theme~="primary"]) [part="content"] ::slotted(vaadin-button) {
```

Comment:
Is there value in these `vaadin-button` rules if they only work for direct children?

Reply:
Yes.

Using `<vaadin-button>` inside is the primary use case. Immediate `<vaadin-button>` is the recommended practice, that way the structure is the simpliest, and gives the most control to notification. We made special efforts here to optimise this, to establish a layout and spacing so that immediate `<vaadin-button>` children look as intended. Indeed there were styles for immediate `<vaadin-button>` children introduced previously in this component.

So theming variants is not first the reason to introduce `<vaadin-button>` styles in the notification. 

Not having theme-variant related overrides of `<vaadin-button>` styles here results in very apparent look-and-feel issues, look at the `primary` notification, for example:
![image](https://user-images.githubusercontent.com/247998/55151222-6e3c6580-5156-11e9-890c-6d006fe64d61.png)
The regular `<vaadin-button>` almost disappears on the blue notification background, and the primary button appears as a regular one.